### PR TITLE
use alpine316

### PR DESCRIPTION
**Pull request by konstruktoid**
This PR:
- replaces the `generic/alpine310` box with `generic/alpine316`, this because Alpine 3.10 has been EOL since 2021-05-01
- performs some basic shell linting

### CODE REVIEW

File: tools/test-setup.sh [hunk 1]
```diff
@@ -85,7 +85,7 @@ if [ -f /etc/os-release ]; then
             # https://github.com/hashicorp/vagrant/issues/11020
             if grep -qi '^CentOS Linux release 8.2.*' /etc/centos-release ; then
                 # https://bugs.centos.org/view.php?id=17120
-                relver="$(cat /etc/centos-release | awk '{print $4}')"
+                relver="$(grep -v '^#' /etc/centos-release | awk '{print $4}')"
```
Comment (apatard) [inline tools/test-setup.sh:88]: Is this a change from shellcheck too ?

**Comment by apatard** [comment]
I had this on my never-ending todo. Thanks for doing it. Why not doing 2 PR : one for the shellcheck and one for the alpine change ? they don't seem related to me but I may be missing something

File: tools/test-setup.sh (hunk 1 above)
Comment (konstruktoid) [reply tools/test-setup.sh:88]: yeah, https://www.shellcheck.net/wiki/SC2002 "Useless cat", but it's really nitpicking in my opinion.

**Comment by konstruktoid** [comment]
> I had this on my never-ending todo. Thanks for doing it. Why not doing 2 PR : one for the shellcheck and one for the alpine change ? they don't seem related to me but I may be missing something

yeah... that seems so obvious when I'm not tired...
I'll create a couple of new and fresh PRs.

**Pull request closed without merging**